code/explainer.py

```diff
@@ -71,47 +72,38 @@ def predict(self, data):
     # ************* pd *************** #
 
     def pd(self, X, idv, grid):
-        """
-        numpy implementation of pd calculation for 1 variable 
-        
-        takes:
-        X - np.ndarray (2d), data
-        idv - int, index of variable to calculate profile
-        
-        returns:
-        y - np.ndarray (1d), vector of pd profile values
-        """
+
+        X = pd.DataFrame(data=X, index=range(X.shape[0]), columns=range(X.shape[1]))
+        # print(X)
+        explainer = dalex.Explainer(self.model, X, verbose=False)
+
+        ld_rf = explainer.model_profile(type = 'accumulated', variables=[idv], grid_points=len(grid), verbose=False)
         
-        grid_points = len(grid)
-        # take grid_points of each observation in X
-        X_long = np.repeat(X, grid_points, axis=0)
-        # take grid for each observation
-        grid_long = np.tile(grid.reshape((-1, 1)), (X.shape[0], 1))
-        # merge X and grid in long format
-        X_long[:, [idv]] = grid_long
-        # calculate ceteris paribus
-        y_long = self.predict(X_long)
-        # calculate partial dependence
-        y = y_long.reshape(X.shape[0], grid_points).mean(axis=0)
+        # print(ld_rf.result)
+
+        output = ld_rf.result['_yhat_'].to_numpy()
+
+        # print(output.shape)
+
+        return output
 
-        return y
 
-    def pd_pop(self, X_pop, idv, grid):
```

Comment:
To wywalam, bo bałem się, że jest gdzieś używane w tym gradientowym i mi się będzie coś liczyć źle, ale wywaliłem i działa, więc chyba jest używane tylko w tym algorytmie genetycznym